toqito/matrices/pauli.py

```diff
@@ -102,13 +102,16 @@ def pauli(ind: int | str | list[int] | list[str], is_sparse: bool = False) -> np
 
     """
     if isinstance(ind, (int, str)):
+        allowed_ind_options = {"x", "X", 1, "y", "Y", 2, "z", "Z", 3, "i", "I", 0}
+        if ind not in allowed_ind_options:
+            raise ValueError(f"Invalid Pauli operator index provided. Allowed options are {allowed_ind_options}.")
         if ind in {"x", "X", 1}:
             pauli_mat = np.array([[0, 1], [1, 0]])
         elif ind in {"y", "Y", 2}:
             pauli_mat = np.array([[0, -1j], [1j, 0]])
         elif ind in {"z", "Z", 3}:
             pauli_mat = np.array([[1, 0], [0, -1]])
-        else:
+        elif ind in {"i", "I", 0}:
```

Comment:
it is not clear why this line is showing up as partially uncovered when there is a unit test for all the options. 

<img width="366" height="155" alt="Image" src="https://github.com/user-attachments/assets/70702e16-d66b-41e2-84ac-20278617d874" />